packages/orchestration/src/orchestration-stream-response.ts

```diff
@@ -41,6 +48,32 @@ export class OrchestrationStreamResponse<T> {
     this._finishReasons = finishReasons;
   }
 
+  public getToolCalls(choiceIndex = 0): MessageToolCalls | undefined {
+    try {
+      const toolCallChunks = this._toolCallChunks.get(choiceIndex);
+      if (!toolCallChunks) {
+        throw new Error(`No tool calls found for choice index ${choiceIndex}`);
+      }
+      const toolCalls: MessageToolCalls = [];
+      for (const chunkArray of toolCallChunks.values()) {
+        const toolCall = mergeToolCallChunks(chunkArray);
```

Comment:
[q] Can we not merge them on the fly? Otherwise if we store chunks, we might as well store chunks also for normal no tool call case? The reason we didn't store chunks is that this could take too much memory. So e.g. for the `toContentStream()` case, we aggregate content already on the fly.

Reply:
Good point, also simplifies the mapping logic.